### PR TITLE
feat: add dataset columns/metrics translation support

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR allows translating columns and metrics names from the datasets. It doesn't affect the queries as the value changed is only the `verbose_name`.

A new dataset is created for every supported language and associated with the corresponding chart.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: scripts/translate_utils.py [hunk 1]
```diff
@@ -3,9 +3,89 @@
...
+    translatable_attributes = []
+
+    def __init__(self, asset: dict):
+        self.asset = asset
+        for key in ASSET_FOLDER_MAPPING:
+            if key in asset:
+                self.asset_type = ASSET_FOLDER_MAPPING[key]
+                break
+
+    def extract_text(self):
```
Comment (bmtcril) [inline scripts/translate_utils.py:16]: Looks like this was left in by mistake?

Comment (Ian2012) [reply scripts/translate_utils.py:16]: yes, I'm trying to make sure all strings are translated as it was before

Comment (Ian2012) [reply scripts/translate_utils.py:16]: Working as before, 236 strings found

File: scripts/translate_utils.py [hunk 2]
```diff
@@ -3,9 +3,95 @@
...
+        return strings
+
+    def translate_var(self, content, var_path: list):
+        """
+        Helper method to remove content from the content dict.
+        """
+        if not content:
+            return []
+        if len(var_path) == 1:
+            # If the content is a list, we need to iterate over each item in the list
```
Comment (bmtcril) [inline scripts/translate_utils.py:38]: I think this comment and the same one below are wrong now? Same with the check below here. They type hint for the function says it's a list? Though that's the only type hint here?

File: scripts/translate_utils.py [hunk 3]
```diff
@@ -3,9 +3,95 @@
...
+            if isinstance(content, list):
+                 strings = []
+                 for item in content:
+                     strings.append(item.get(var_path[0], ""))
+                 return strings
+             # print("Translating asset: ", self.asset_type, "var_path: ", var_path)
```
Comment (bmtcril) [inline scripts/translate_utils.py:44]: You can delete this if it's not being used.

File: scripts/translate_utils.py [hunk 4]
```diff
@@ -3,9 +3,95 @@
...
+            string = [content.get(var_path[0], "")]
+            return string or []
+        else:
+            # If the content is a list, we need to iterate over each item in the list
```
Comment (bmtcril) [inline scripts/translate_utils.py:48]: Same comments as above, I'm confused if this is a list or not, or when it might be one instead of the other.

**Review by bmtcril — COMMENTED** [review]
Looking good, just a couple of questions

File: scripts/translate_utils.py (hunk 1 above)
Comment (bmtcril) [reply scripts/translate_utils.py:16]: Yes, but why is it defined twice? Here and on line 21?

**Review by bmtcril — APPROVED** [review]
👍 nice!

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-aspects&entry.867055334=https://github.com/openedx/tutor-contrib-aspects/pull/641&entry.1484655318=https://github.com/Ian2012&entry.752974735=2024-03-07+15:01&entry.1917517419=2024-03-11+19:44&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**